ui/de-lib/src/main/java/org/iplantc/de/admin/desktop/client/ontologies/presenter/OntologiesPresenterImpl.java

```diff
@@ -449,4 +462,40 @@ public OntologyHierarchy getSelectedHierarchy() {
         return view.getSelectedHierarchy();
     }
 
+    @Override
+    public void onDeleteOntologyButtonClicked(final DeleteOntologyButtonClickedEvent event) {
+        serviceFacade.deleteOntology(event.getOntologyVersion(), new AsyncCallback<Void>() {
+            @Override
+            public void onFailure(Throwable caught) {
+                ErrorHandler.post(caught);
+            }
+
+            @Override
+            public void onSuccess(Void result) {
+                announcer.schedule(new SuccessAnnouncementConfig(appearance.ontologyDeleted(event.getOntologyVersion())));
+                getOntologies(false);
+            }
+        });
+    }
+
+    @Override
+    public void onDeleteHierarchy(final DeleteHierarchyEvent event) {
+        List<OntologyHierarchy> deletedHierarchies = event.getDeletedHierarchies();
+        for (final OntologyHierarchy hierarchy : deletedHierarchies) {
```

Comment:
Yes 👍 I just posted a comment explaining this elsewhere seconds before you asked this question :)

I think the documentation for the hierarchy delete endpoint needs to be clearer about this, so I'll also update the docs when I fix the bug.